src/meta/src/rpc/server.rs

```diff
@@ -341,64 +345,166 @@ pub async fn rpc_serve_with_store<S: MetaStore>(
 // Print the sleep intervals
 // use pseudo rand gen with seed
 mod tests {
+    use std::net::{IpAddr, Ipv4Addr};
+
+    use risingwave_pb::common::WorkerType;
+    use risingwave_rpc_client::MetaClient;
     use tokio::time::sleep;
 
     use super::*;
 
+    async fn setup_n_nodes(n: u16) -> Vec<(JoinHandle<()>, WatchSender<()>)> {
+        let meta_store = Arc::new(MemStore::default());
+
+        let mut node_controllers: Vec<(JoinHandle<()>, WatchSender<()>)> = vec![];
+        for i in 0..n {
+            // TODO: add pseudo random sleep here
+            let node = format!("node{}", i);
+            let err_msg = format!("Meta {} failed in setup", node);
+            let err_msg = err_msg.as_str();
+
+            let info = AddressInfo {
+                addr: node,
+                listen_addr: SocketAddr::new(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), 1234 + i),
+                prometheus_addr: None,
+                dashboard_addr: None,
+                ui_path: None,
+            };
+            node_controllers.push(
+                rpc_serve_with_store(
+                    meta_store.clone(),
+                    info,
+                    Duration::from_secs(4), // What values do we want to use here
+                    2,                      // What values do we want to use here
+                    MetaOpts::test(true),   // True or false?
+                )
+                .await
+                .expect(err_msg),
+            );
+        } // sleep duration of election cycle, not fixed duration
+        sleep(Duration::from_secs(10)).await;
+        node_controllers
+    }
+
     #[tokio::test]
-    async fn test_leader_lease() {}
-}
+    async fn test_single_leader_setup() {
+        // make n also pseudo random?
+        let n = 5;
+        let _node_controllers = setup_n_nodes(n).await;
+
+        let mut leader_count = 0;
+        for i in 0..n {
+            // create client connecting against meta_i
+            let port = 5678 + i;
+            let meta_addr = format!("127.0.0.1:{}", port);
```

Comment:
🥲 You started meta with port from 1234, but used 5678 here.

Reply:
Yap. You are absolutely correct.